free-1.md

```diff
@@ -10,34 +10,84 @@ _What need is met by your idea? Who are you helping? What is that person's
+to generate high level flowcharts that express the intended algorithm.
+
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+1. The programmer will likely not want every implementation detail to leak into the flowcharts,
+so there should be some way to indicate what _should_ appear on the flowcharts.
+
+2. There should be as few barriers as possible to using this tool,
+so I think allowing users to simply add code or annotations to the working code would be ideal.
+
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+I really like the idea of making algorithms represented in code more accessible.
+I myself tend to favor and think in visualizations, and I think it would be a lot easier to explain code to someone
+if I could easily transform my code into a flowchart. During my internship over the summer,
+I wrote a document that details the design of my project, and in that document,
+I created a flowchart to represent how I decided to handle one of the more complex parts of the project
+because a flowchart seemed fitting for the task. I also think documentation is _super_ important,
+so it's very appealing to me to create tools that ameliorate the documentation process
+and compel others to put more effort toward it.
+
 
 ### Domain
 _Describe the project's domain in five words._
 
+Code Documentation and Visualization
+
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+Again, I'd like the code to feel as much like the underlying language as possible
+so that there's not too much of a learning curve and to minimize the barriers
+that might cause one to avoid using the tool. To that end,
+my initial thoughts go to [Javadoc] annotations and [Doxygen] comments,
+which use minimal indicators in the code to denote something.
+I'm leaning toward comments for the syntax because I think they will be less offensive and confusing than a Java-style annotation.
+
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+When a program runs, the marked up file is parsed, and one or more flowcharts are created based on the markings.
+I think it would be really cool if users could interact with the generated flowcharts:
+for instance, one of the blocks that indicates some sort of helper function
+can be clicked to reveal the helper function's algorithm in the form of another flowchart.
+If the proposed language is mostly text replacement, the synax _is_ based on comments,
+and the flowchart structure largely comes from the code, then I don't foresee a need for very many errors;
+however, I think errors _can_ be communicated on the command line.
+
```

Comment:
The flowcharts would be generated by parsing the comments along with the GPL control flow code (i.e., conditionals and loops).

I'm not certain I understand the second question. I think comments that should be used in the flowchart generation would use some indicator like `//$` instead of just `//` in languages like C++ and Java (Flowgen uses this syntax). I think it would also be nice to support adding comments to the flowcharts though.